src/App.jsx

```diff
@@ -14,33 +15,91 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const initialGoods = [...goodsFromServer];
+  const [goods, setGoods] = useState([...initialGoods]);
+  const [isReversed, setIsReversed] = useState(false);
+  const [isAlpha, setIsAlpha] = useState(false);
+  const [isSortByLength, setIsSortByLength] = useState(false);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const sortAlphabetically = () => {
+    const sortedGoods = [...goods].sort((a, b) => a.localeCompare(b));
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    setGoods(sortedGoods);
+    setIsAlpha(true);
+    setIsSortByLength(false);
+    setIsReversed(false);
+  };
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  const sortByLength = () => {
+    const sortedGoods = [...goods].sort((a, b) => a.length - b.length);
+
+    setGoods(sortedGoods);
+    setIsSortByLength(true);
+    setIsAlpha(false);
+    setIsReversed(false);
+  };
+
+  const reverseGoods = () => {
+    const sortedGoods = [...goods].reverse();
+
+    setGoods(sortedGoods);
+    setIsReversed(!isReversed);
+  };
+
+  const resetGoods = () => {
+    setGoods([...goodsFromServer]);
+    setIsAlpha(false);
+    setIsSortByLength(false);
+    setIsReversed(false);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const isInitialOrder = JSON.stringify(goods) === JSON.stringify(initialGoods);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${isAlpha} ? '' : 'is-light'`}
```

Comment:
The template literal for the className is incorrect. The condition should be inside the curly braces, like this: `${isAlpha ? '' : 'is-light'}`.